sklearn/feature_extraction/image.py

```diff
@@ -338,6 +338,18 @@ def _extract_patches(arr, patch_shape=8, extraction_step=1):
     return patches
 
 
+@validate_params(
+    {
+        "image": [np.ndarray],
```

Comment:
@jeremiedbb Since we have only `np.ndarray` here, the `generate_valid_param` will try to generate a NumPy array by calling `np.ndarray.type()`. However, it will raise:

```pytb
TypeError: ndarray() missing required argument 'shape' (pos 1)
```

In other cases, we have as well `None` and I assume it was the value generated. Do you think that we should specialize the code by generating a NumPy array with a given shape? Do you forsee an issue if the shape of the array is not the right one (I don't think so since we should not execute the function code indeed).